tests/push/test_ami_push.py

```diff
@@ -229,6 +278,42 @@ def test_push_public_image(command_tester):
         ],
     )
 
+    # Check that aws publish has been called twice
```

Comment:
I'm a little confused about a few things for this test:

- mock_aws_publish was called twice - OK
- only the first call's metadata is checked - could be OK, as it's not required that every test covers every single detail.  But I'm wondering, if it's called twice with different arguments, is it guaranteed which call happens first?  If not, this test could be flaky.
- the collector items and images.json contain only a single item - how does that connect with mock_aws_publish being called twice?  Is it called twice because, for example, it's retrying a publish of a single item?  But I don't see any setup to simulate an error.

Reply:
The aws publish was called twice, because once we pushed an image with groups set to their normal values (see line [127](https://github.com/release-engineering/pubtools-ami/blob/b0e66bfe450f36b3584b29a3faf1c1ee246c58f0/pubtools/_ami/tasks/push.py#L127)) and once we push the public image with groups set to `all` (see line [148](https://github.com/release-engineering/pubtools-ami/blob/b0e66bfe450f36b3584b29a3faf1c1ee246c58f0/pubtools/_ami/tasks/push.py#L148)). Since incidentally the image had groups set to `all` from the start of the test, the two calls were identical. (And since the collector stores the uploaded items as a dictionary, we got one images.json as well.)

I have added an assert to make sure the calls were really identical to make the test more robust.